### PR TITLE
Expand fragment class tests

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: json_document/tests/test_fragment_class.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+"""Unit tests for fragment classes."""
+
+from unittest2 import TestCase
+
+from json_document.document import DocumentFragment
+
+
+class DocumentFragmentClassTests(TestCase):
+    """Tests related to fragment classes."""
```
Comment (zyga) [inline json_document/tests/test_fragment_class.py]: Per PEP257 there should be a newline before the docstring here:

```
class foo:

    """foo is fooing."""

    ...
```

File: json_document/tests/test_fragment_class.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    def test_getitem_uses_sub_value_class_from_object_property(self):
+
+        class SpecialDocumentFragment(DocumentFragment):
+            pass
+
+        fragment = DocumentFragment(
+            document=None,
+            parent=None,
+            value={
+                "item": "value"},
```
Comment (zyga) [inline json_document/tests/test_fragment_class.py]: Let's keep that on one line: ```value={"item": "value"},```

**Comment by zyga** [comment]
Thanks a lot! Everything looks good. I just commented on a few minor stylistic issues. I'll merge this if you can fix them.

File: json_document/tests/test_fragment_class.py (hunk 2 above)
Comment (jayvdb) [inline json_document/tests/test_fragment_class.py]: This is copying the code from `test_misc.py`, and changing this line with break the blame of `git blame -wC` .

While I am modifying the method name, I'd prefer that the blame of the body of the method doesnt mis-attribute it to me.

File: json_document/tests/test_fragment_class.py (hunk 1 above)
Comment (jayvdb) [inline json_document/tests/test_fragment_class.py]: That depends on the flavour.  There are two conflicting error codes, due to the pep 257 changing.

    D203: 1 blank line required before class docstring

    D211: No blank lines allowed before class docstring

If we want to follow the current pep, we would need to disable D203, and then remove blank lines before the class docstring.

However if you prefer the old pep, I am more than happy to add a blank line.  My other projects usually take this approach, as I find that extra blank line helpful.

Comment (zyga) [inline json_document/tests/test_fragment_class.py]: Eh, standards :-) Let's ignore this. I might do a global cleanup later.

File: json_document/tests/test_fragment_class.py (hunk 2 above)
Comment (zyga) [inline json_document/tests/test_fragment_class.py]: Fine, +1

**Pull request merged**